resteasy-core/src/main/java/org/jboss/resteasy/plugins/providers/RegisterBuiltin.java

```diff
@@ -30,6 +35,39 @@
  */
 public class RegisterBuiltin
 {
+   private static final Map<ClassLoader, ResteasyProviderFactory> configuredClientFactories = new WeakHashMap<>();
+   private static final boolean gzipForCachedFactories = isGZipEnabled();
+
+   public static synchronized ResteasyProviderFactory getClientInitializedResteasyProviderFactory(ClassLoader cl)
+   {
+      ResteasyProviderFactory rpf = null;
+      final boolean gzip = isGZipEnabled();
+      if (gzipForCachedFactories == gzip) {
```

Comment:
When this can be false? (I'm probably missing some information ...)

Reply:
it can be false in case the resteasy.allowGzip property is changes after this class has been loaded